File: python/abc235_c.py
from collections import Counter

n, q = map(int, input().split())
a = list(map(int, input().split()))
xk = [list(map(int, input().split())) for _ in range(q)]

map = Counter()
ans = {}

for i in range(n):
    map[a[i]] += 1
    ans[(a[i], map[a[i]])] = i + 1

for x, k in xk:
    if (x, k) in ans:
        print(ans[(x, k)])
    else:
        print(-1)

File: rust/others/src/bin/abc235_c.rs
use proconio::input;
use std::collections::HashMap;

fn main() {
    input! {
        n: usize,
        q: usize,
        a: [usize; n],
        xk: [(usize, usize); q],
    }
    let mut map = HashMap::new();
    let mut ans = HashMap::new();
    for i in 0..n {
        *map.entry(a[i]).or_insert(0usize) += 1usize;
        let key: (usize, usize) = (a[i], map[&a[i]]);
        ans.insert(key, i + 1);
    }
    for i in 0..q {
        let key = xk[i];
        if ans.contains_key(&key) {
            println!("{}", ans[&key]);
        } else {
            println!("-1");
        }
    }
}
